api/server/httputils/errors.go

```diff
@@ -37,49 +33,45 @@ func GetHTTPErrorStatusCode(err error) int {
 	}
 
 	var statusCode int
-	errMsg := err.Error()
 
-	switch e := err.(type) {
-	case httpStatusError:
-		statusCode = e.HTTPErrorStatusCode()
-	case inputValidationError:
+	// Stop right there
+	// Are you sure you should be adding a new error class here? Do one of the existing ones work?
+
+	// Note that the below functions are already checking the error causal chain for matches.
+	switch {
```

Comment:
The problem with this approach is it's treating errors a bit like exceptions where the caller must know every kind of thing that it could happen from the lower layer.
In both cases untyped cases can escape, but in the case of the scope of these escapes is limited to just the things that the lower layer didn't expect (not sure if that wording makes sense...).